lib-injection/sitecustomize.py

```
@@ -1,57 +1,29 @@
-    os.environ["PYTHONPATH"] = "%s%s%s" % (bootstrap_dir, os.path.pathsep, prev_python_path)
-
-    # Also insert the bootstrap dir in the path of the current python process.
-    sys.path.insert(0, bootstrap_dir)
-    print("datadog autoinstrumentation: successfully configured python package")
-
-
-# Avoid infinite loop when attempting to install ddtrace. This flag is set when
-# the subprocess is launched to perform the installation.
-if "DDTRACE_PYTHON_INSTALL_IN_PROGRESS" not in os.environ:
-    try:
-        import ddtrace  # noqa: F401
-
-    except ImportError:
-        import subprocess
-
-        print("datadog autoinstrumentation: installing python package")
+    os.environ["PYTHONPATH"] = "%s%s%s" % (path, os.pathsep, prev_python_path)
+    sys.path.insert(0, path)
 
-        # Set the flag to avoid an infinite loop.
-        env = os.environ.copy()
-        env["DDTRACE_PYTHON_INSTALL_IN_PROGRESS"] = "true"
 
-        if "git" in version:
-            ddtrace_version = version
-        else:
-            ddtrace_version = "ddtrace==%s" % version
+pkgs_path = os.path.join(os.path.dirname(__file__), "ddtrace_pkgs")
+bootstrap_dir = os.path.join(pkgs_path, "ddtrace", "bootstrap")
+_add_to_pythonpath(pkgs_path)
+_add_to_pythonpath(bootstrap_dir)
 
-        # Execute the installation with the current interpreter
-        try:
-            subprocess.run([sys.executable, "-m", "pip", "install", ddtrace_version], env=env, check=True)
-        except Exception:
-            print("datadog autoinstrumentation: failed to install python package version %r" % ddtrace_version)
-        else:
-            print("datadog autoinstrumentation: successfully installed python package version %r" % ddtrace_version)
-            _configure_ddtrace()
-    else:
-        print("datadog autoinstrumentation: ddtrace already installed, skipping install")
-        _configure_ddtrace()
+try:
+    import ddtrace.bootstrap.sitecustomize  # noqa: F401
+except ImportError as e:
```

Should we attempt to catch all exceptions that could be raised by configuring the tracer? This script could attempt to install a bad release.

---

:+1: Not a bad idea, there's no harm here in catching everything to avoid crashing the application.